src/include/sof/library_manager.h

```diff
@@ -0,0 +1,115 @@
+/* SPDX-License-Identifier: BSD-3-Clause
```

Comment:
Can you add some more details about how this works.

- how is the memory for loading the lib is managed. Who reserves it? 
   - it looks like the module is binary is sent by the Host AP and DSP parses and loads it. Is that right?
- lets try to make this as Intel agnostic as possible. It should be a generic functionality. So maybe avoid terms like IMR / HPSRAM.

Reply:
Memory manager is being prepared and will be added soon as a part of a FW. I will consider memory names to be more universal.